src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -15,30 +22,66 @@
   /** Store class that gives access to Users. */
   private UserStore userStore;
 
+  /** Store class that gives access to Conversations. */
+  private ConversationStore conversationStore;
+
   /** Set up state for handling activty feed requests. */
   @Override
   public void init() throws ServletException {
   	super.init();
     setUserStore(UserStore.getInstance());
+    setConversationStore(ConversationStore.getInstance());
   }
 
   /**
-   * Sets the UserStore used by this servlet. This function provides a common setup method for use
-   * by the test framework or the servlet's init() function.
+   * Sets the UserStore used by this servlet. This function provides a common 
+   * setup method for use by the test framework or the servlet's init() function.
    */
   void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
 
+  /**
+   * Sets the ConversationStore used by this servlet. This function provides a 
+   * common setup method for use by the test framework or the servlet's init() function.
+   */
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+
   /**
    * This function fires when a user navigates to the activity feed page. It 
    * simply forwards to activityfeed.jsp for rendering.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) 
       throws IOException, ServletException {
+    // Get and set Users
     List<User> users = userStore.getAllUsers();
     request.setAttribute("users", users);
+
+    // Get and set Conversations
+    List<Conversation> conversations = conversationStore.getAllConversations();
+    request.setAttribute("conversations", conversations);
+
+    // Make, sort, and set the event map
+    Map<Instant, Object> unsortedEventsMap = new HashMap<Instant, Object>();
+    for(User user : users)
+      unsortedEventsMap.put(user.getCreationTime(), user);
+    for(Conversation conversation : conversations)
+      unsortedEventsMap.put(conversation.getCreationTime(), conversation);
+    request.setAttribute("unsortedEventsMap", unsortedEventsMap);
+    // Sort the event map by key in descending order
+    Map<Instant, Object> sortedEventsMap = 
+      new TreeMap<Instant, Object>(new Comparator<Instant>() {
+        @Override
+        public int compare(Instant o1, Instant o2) {
+          return o2.compareTo(o1);
+        }
+      });
+    sortedEventsMap.putAll(unsortedEventsMap);
+    request.setAttribute("sortedEventsMap", sortedEventsMap);
```

Comment:
If the above comment is correct and all we need is the sorted map, how about we create it and insert items into it directly?

Reply:
Will do.

Reply:
you could do that initially when all the users and conversations are loaded so it doesn't do it on every request.

Reply:
Sorry I'm not sure what you mean by this. Do you mean having it in init()? @vocatwk